src/storage/v2/edge_accessor.cpp

```diff
@@ -70,51 +68,112 @@ bool EdgeAccessor::IsVisible(const View view) const {
         case Delta::Action::DELETE_DESERIALIZED_OBJECT:
         case Delta::Action::DELETE_OBJECT:
           break;
-        case Delta::Action::ADD_OUT_EDGE: {  // relevant for the from_vertex_ -> we just deleted the edge
+        case Delta::Action::ADD_OUT_EDGE: {
+          if (delta.vertex_edge.edge == edge_ && delta.vertex_edge.edge_type == edge_type_) {
+            attached = true;
+          }
+          break;
+        }
+        case Delta::Action::REMOVE_OUT_EDGE: {
```

Comment:
Check for edge type here as well?

Reply:
Do we have to traverse the whole chain?
Is there a case where an edge has been deleted, but could reaper afterwards?
I guess if you are changing the type back and forth?

Reply:
why do we want to check edge here? it has been detached from the vertex. it's not part of the vertex

Reply:
Maybe I misunderstood how EdgeTypeChange works, but, in that case, wouldn't we have a remove edge and than add edge with the same gid, just different types. So an edge could still be attached even if there's a REMOVE delta.

Reply:
`wouldn't we have a remove edge and than add edge with the same gid, just different types.`

yes, that's the situation now

Reply:
So why don't we need to check the type?

Reply:
Fine as is. The type check would be redundant.